### PR TITLE
[feature/88865] Implemented scratchpad

### PR DESCRIPTION
**Pull request by iTweeno**
Link to Shortcut: https://app.shortcut.com/deskpro/story/88865/scratchpad-app-mvp

### CODE REVIEW

File: src/pages/Main.tsx [hunk 1]
```diff
@@ -1,7 +1,58 @@
...
 export const Main = () => {
-    return (
-        <Button text="Greet" onClick={() => alert("Hello")} />
-    );
+  const { client } = useDeskproAppClient();
+
+  const [text, setText] = useState<string>("");
+  const debouncedText = useDebounce(text, 200);
+
+  useEffect(() => {
```
Comment (AshleyDawson) [inline src/pages/Main.tsx:14]: We've got a new way of doing this using a special hook. It basically does what you're doing here but encapsulates the function in a test for an initialized Deskpro apps client. You can see an example of it here: https://support.deskpro.com/hr-HR/guides/developers/how-to-use-state-to-store-notes#anchor-1

So this code would change to something like:

```typescript
useInitialisedDeskproAppClient((client) => {
  client.setState<string>("scratchpad", debouncedText);
}, [debouncedText]);
```

File: src/pages/Main.tsx [hunk 2]
```diff
@@ -1,7 +1,58 @@
...
+    client?.setState<string>("scratchpad", debouncedText);
+  }, [debouncedText]);
+
+  useEffect(() => {
+    client
+      ?.getState<string>("scratchpad")
```
Comment (AshleyDawson) [inline src/pages/Main.tsx:20]: Please change this to use `useInitialisedDeskproAppClient()` too.

Also, we need to use `setUserState()` and `getUserState()` as `setState()` will store these notes globally and we want to store them against the user, see: https://app.shortcut.com/deskpro/story/88865/scratchpad-app-mvp

**Review by AshleyDawson — CHANGES_REQUESTED** [review]

File: src/pages/Main.tsx [hunk 3]
```diff
@@ -1,7 +1,70 @@
...
+    client
+      ?.getUserState<string>("scratchpad")
+      .then((res: GetStateResponse<string>[]) => {
+        setText(res[0]?.data ?? "");
+      });
+  });
+
+  useInitialisedDeskproAppClient(
+    (client) => {
+      debouncedText &&
```
Comment (AshleyDawson) [inline src/pages/Main.tsx:30]: Will this still work if the user deletes all of the text from the textarea? I.e. they select all of the text and press backspace.

We want to allow the user to manually delete all content from the textarea and the app save an empty string in this case.

Comment (iTweeno) [reply src/pages/Main.tsx]: Please check the latest commit

**Review by AshleyDawson — APPROVED** [review]

**Pull request merged**